tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/analyses/ExpansionMarker.scala

```diff
@@ -46,7 +47,10 @@ class ExpansionMarker @Inject() (tracker: TransformationTracker) extends TlaExTr
       val tag = ex.typeTag
       if (shallExpand) {
         // Expand everything, including the function set.
-        logger.warn(s"The set $ex will be expanded. This will blow up the solver.")
+        logger.warn(
+            s"The function set [ S -> T ] = $ex will be expanded, producing Cardinality(T)^Cardinality(S) constraints.")
+        logger.warn(
+            "Expansion may significantly slowdown the solver, e.g., when Cardinality(T) >= 4 or Cardinality(S) >= 5.")
```

Comment:
Given that this depends on both base and exponent, should it be "and"?

```suggestion
            "Expansion may significantly slowdown the solver, e.g., when Cardinality(T) >= 4 and Cardinality(S) >= 5.")
```

Reply:
On a second thought, seems like you want to indicate that things slow down above ~1000 constraints. That's somewhat lost by just giving an example; should we specify that number explicitly?

```suggestion
            "Expansion may significantly slowdown the solver, e.g., when Cardinality(T)^Cardinality(S) >= 1000.")
```